src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,65 @@
+package duke;
+
+/**
+ * The Ui class is responsible for handling all user interface elements
+ * of the application, such as greeting users, displaying messages, and showing errors.
+ * It provides methods to print various types of messages to the console.
+ */
+public class Ui {
+    /**
+     * Displays a greeting message to the user.
+     * This includes an ASCII art representation of a snowman and a welcome message.
+     */
+    public static String greet() {
+        return " Hello! I'm SnowBoy\n" + " What can I do for you?";
+    }
+
+    /**
+     * Displays a loading error message.
+     * This is typically called when the application fails to load an existing task list
+     * and needs to create a new one.
+     */
+    public void showLoadingError() {
+        System.out.println(" No existing list detected. Creating new list...");
+    }
+
+    /**
+     * Displays an exit message to the user.
+     * This message is shown when the user decides to exit the application.
+     */
+    public static String exit() {
+        return " Bye. Hope to see you again soon!";
+    }
+
+    /**
+     * Processes the user command and directs it to the appropriate method.
+     * Supports a variety of commands such as adding, deleting, and listing tasks.
+     *
+     * @param tl The TaskList to perform operations on.
+     * @param cmd The user input command to be processed.
+     * @throws DukeException If the command is invalid or incorrectly used.
+     */
+    public static String checkCmd(TaskList tl, String cmd) throws DukeException {
```

Comment:
Perhaps more contextual argument names
```suggestion
    public static String checkCmd(TaskList tasks, String command) throws DukeException {
```